File: 0100-same-tree/0100-same-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        vector<int> pvalues;
        vector<int> pdirect;
        inorder(p,pvalues,pdirect,0);

        vector<int> qvalues;
        vector<int> qdirect;
        inorder(q,qvalues,qdirect,0);

        return (pvalues==qvalues) && (pdirect==qdirect);
    }

    void inorder(TreeNode* root, vector<int>& nums, vector<int>& direct, int direction){
        if(root==nullptr){
            return;
        }
        inorder(root->left,nums,direct,-1);
        nums.emplace_back(root->val);
        direct.emplace_back(direction);
        inorder(root->right,nums,direct,1);
    }
};

File: 0100-same-tree/0100-same-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    Stack<String> treeP;
    Stack<String> treeQ;
    public boolean isSameTree(TreeNode p, TreeNode q) {
        treeP = new Stack<>();
        addToList(p,treeP);
        treeQ = new Stack<>();
        addToList(q,treeQ);
        return treeP.equals(treeQ);
    }

    public void addToList(TreeNode p, Stack<String> treeP){
        if(p==null){
            treeP.push("null");
            return;
        }else{
            treeP.push(Integer.toString(p.val));
        }
        addToList(p.left,treeP);
        addToList(p.right,treeP);
    }
}